### DISCUSSION TITLE
Change Marker type

### DISCUSSION THREAD

**Discussion by automartin5000** [projen/projen#3319] [Q&A] [answered]
I have a Groovy file I want to mark as generated. If I use a TextFile, I'm guessing it'll mark it with a `#` instead of `//`. Is there a way to change the marker type?

**Comment by mrgrain** [comment, marked as answer]
`TextFile` doesn't add a marker by itself. If you want to hook into the existing infrastructure for it, you'll need to check `this.marker` inside of `synthesizeContent()` and add it in the appropriate location. Most notably, `this.marker` does NOT include any kind of comment syntax. It's the text only.

A simple implementation would look like this:
```ts
export class GroovyFile extends TextFile {
  protected synthesizeContent(resolver: IResolver): string | undefined {
    if (this.marker) {
      this.lines.unshift(`// ${this.marker}`);
    }
    return super(resolver);
  }
}
```

**Reply by automartin5000** [reply]
Perfect thanks!
